compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3224,10 +3229,113 @@ public void visit(BLangUnaryExpr unaryExpr) {
         resultType = types.checkType(unaryExpr, actualType, expType);
     }
 
+    public void visit(BLangAnnotationAttachment annAttachmentNode) {
```

Comment:
I'm not sure if this belongs here. Why are we moving this from SemanticAnalyzer, given that `BLangAnnotationAttachment` itself is not an expression, and only the `expr` part is?

We already seem to be doing the relevant semantic/constant analyzer based analysis in SemanticAnalyzer, and only seem to be calling into TypeChecker for the expression https://github.com/ballerina-platform/ballerina-lang/pull/23829/files#diff-7123076514bbb67aa85076709d29ee0fL2880.

Reply:
What I want to do is execute annotation validation logic from `typecheker` since that's where we see the annotation on a `typeconvertion` expression.

If not moving to typechecker we will have to move that logic to its own class, but I don't see any compelling advantage of moving to a separate class that moving to typecheker.

Reply:
It seems we do call back to `SemanticAnalyzer` from `typechecker`.

Reply:
Yeah, there was a similar requirement with streaming and it was decided to have this circular dependency for the time-being instead of moving the logic.